src/formatting/modules.rs

```diff
@@ -122,14 +164,11 @@ impl<'ast, 'sess, 'c> ModResolver<'ast, 'sess> {
             _ => PathBuf::new(),
         };
 
-        // Skip visiting sub modules when the input is from stdin.
-        if self.recursive {
-            self.visit_mod_from_ast(&krate.module)?;
-        }
```

Comment:
IIUC, this conditional is dropped because we need to parse the mods even when `recursive` is false in order to be able to see the inner attributes on the mods so that reordering can be done safely, right?

Reply:
Yes, that's correct.